datasets/docred/docred.py

```diff
@@ -0,0 +1,119 @@
+entity pairs. In order to accelerate the research on document-level RE, we introduce DocRED, a new dataset constructed \
+from Wikipedia and Wikidata with three features:
+    - DocRED annotates both named entities and relations, and is the largest human-annotated dataset for document-level RE from plain text.
+    - DocRED requires reading multiple sentences in a document to extract entities and infer their relations by synthesizing all information of the document.
+    - Along with the human-annotated data, we also offer large-scale distantly supervised data, which enables DocRED to be adopted for both supervised and weakly supervised scenarios.
+"""
+
+_URLS = {
+    "dev": "https://drive.google.com/uc?export=download&id=1fDmfUUo5G7gfaoqWWvK81u08m71TK2g7",
+    "train_distant": "https://drive.google.com/uc?export=download&id=1fDmfUUo5G7gfaoqWWvK81u08m71TK2g7",
+    "train_annotated": "https://drive.google.com/uc?export=download&id=1NN33RzyETbanw4Dg2sRrhckhWpzuBQS9",
+    "test": "https://drive.google.com/uc?export=download&id=1lAVDcD94Sigx7gR3jTfStI66o86cflum",
+    "rel_info": "https://drive.google.com/uc?id=1y9A0zKrvETc1ddUFuFhBg3Xfr7FEL4dW&export=download",
+}
+
+
+class DocRed(nlp.GeneratorBasedBuilder):
+    """DocRED: A Large-Scale Document-Level Relation Extraction Dataset"""
+
+    def _info(self):
+        return nlp.DatasetInfo(
+            description=_DESCRIPTION,
+            features=nlp.Features(
+                {
+                    "title": nlp.Value("string"),
+                    "sents": nlp.features.Sequence(nlp.features.Sequence(nlp.Value("string"))),
+                    "vertexSet": [[{
+                         "name": nlp.Value("string"),
+                          "sent_id": nlp.Value("int32"),
+                          "pos": nlp.features.Sequence(nlp.Value("int32")),
+                         "type": nlp.Value("string"),
+                     }]],
+                    "labels": nlp.features.Sequence(
+                        {
+                            "head": nlp.Value("int32"),
+                            "tail": nlp.Value("int32"),
+                            "relation_id": nlp.Value("string"),
+                            "relation_text": nlp.Value("string"),
+                            "evidence": nlp.features.Sequence(nlp.Value("int32")),
+                        }
+                    ),
+                }
+            ),
+            supervised_keys=None,
+            homepage="https://github.com/thunlp/DocRED",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        downloads = {}
+        for key in _URLS.keys():
+            downloads[key] = dl_manager.download_and_extract(_URLS[key])
+            #  Fix for dummy data
+            if os.path.isdir(downloads[key]):
+                downloads[key] = os.path.join(downloads[key],key+".json")
```

Comment:
You can just do `downloads = dl_manager.download_and_extract(_URLS)`.
By doing that, the trick for dummy data won't be necessary.